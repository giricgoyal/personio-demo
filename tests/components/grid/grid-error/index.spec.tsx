import React from 'react'
import GridError from 'src/components/grid/grid-error'
import { render } from 'tests/test-utils'

describe('components/grid/grid-error', () => {
    let container, onGridRefreshHandler, rerender

    beforeEach(() => {
        onGridRefreshHandler = jest.fn()
        ;({ container, rerender } = render(<GridError error="an error" onGridRefresh={onGridRefreshHandler} />))
    })

    test('should render and match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('When no error', () => {
        beforeEach(() => {
            rerender(<GridError onGridRefresh={onGridRefreshHandler} />)
        })

        test('should not render and match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })
})
