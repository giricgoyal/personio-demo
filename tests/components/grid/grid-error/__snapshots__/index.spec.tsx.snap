// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid/grid-error When no error should not render and match snapshot 1`] = `<div />`;

exports[`components/grid/grid-error should render and match snapshot 1`] = `
<div>
  <div
    class="grid__error"
  >
    an error
    <div
      class="grid__error__refresh"
    >
      Click
      <a
        data-testid="refresh-grid-button"
      >
        <i
          class="fas fa-sync fa-2x"
        />
      </a>
      to refresh.
    </div>
  </div>
</div>
`;
